### DISCUSSION TITLE
Publish list of available environment variables

### DISCUSSION THREAD

**Discussion by jonfreynik** [PowerDNS/pdns#13328] [Q&A] [answered]
- Program: Authoritative 
 - Issue type: Feature request

### Short description

Interested in a published list of the different override-able environment variables for docker-compose.yml
### Usecase

Easier ability to work with docker-compose.
### Description

It would be great if there was a list that showed the available environment variables for docker-compose. It seems every online example I can find uses different variable names, and I'm not sure where they are coming up with the naming convention because it does not look to match.

**Comment by phonedph1** [comment]
https://github.com/PowerDNS/pdns/blob/master/Docker-README.md
https://github.com/PowerDNS/pdns/blob/master/docker-compose.yml

Are these helpful at all?

**Comment by jonfreynik** [comment]
Thank you but I'm specifically looking for which env variables are used to link PDNS to mysql instead of the default SQLite database - so far I've been unsuccessful overriding the .conf file with GMYSQL variables and I'm having trouble finding the correct variables names to use - I've searched the code and come up with things like:

gmysql-host=...
gmysql-password=...

but it does not seem to work - I will be reviewing the logs shortly but wanted to put out a feature request for the environment variables because that would have helped me immensely - or maybe I'm missing a good way to find what they should be (I'm still pretty new to docker-compose).

**Reply by Habbie** [reply, marked as answer]
Those are not environment variables. They need to go in `pdns.conf` or on the commandline (with `--` prefixed)

**Reply by jonfreynik** [reply]
Appreciate it, thanks @Habbie

**Comment by Habbie** [comment]
Hello,

our GitHub issue tracker is for bug reports and feature requests. Your question looks like a support question. Support questions are handled in our other online communities: [GitHub Discussions](https://github.com/PowerDNS/pdns/discussions), IRC, and our mailing lists. Please see https://www.powerdns.com/opensource.html for information about those.

I'm moving this ticket to GitHub Discussions.
